[Liu Zhilan] iP

LiuZhi1an opened the pull request:
Duke Task Manager

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

- text-based
- easy to learn
- ~~fast~~ _super fast_ to use

All you need to do is,

1. download it from [here](https://github.com/LiuZhi1an/ip.git).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😏

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing todos
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public static void main(String[] args) {
        new Duke("data/tasks.txt").run();
}
```

LiuZhi1an committed:
Level-1

LiuZhi1an committed:
Level-2

LiuZhi1an committed:
Level-2

LiuZhi1an committed:
Level-2

LiuZhi1an committed:
Level-3

LiuZhi1an committed:
Level-4

LiuZhi1an committed:
A-TextUiTesting

LiuZhi1an committed:
changed static methods

LiuZhi1an committed:
Level-5

LiuZhi1an committed:
change from array to arraylist

LiuZhi1an committed:
Level-6

LiuZhi1an committed:
Level-6

LiuZhi1an committed:
added javadoc

LiuZhi1an committed:
updated expected output

LiuZhi1an committed:
Level-7

LiuZhi1an committed:
Level-8 basics

LiuZhi1an committed:
Delete commented out lines

LiuZhi1an committed:
completed Level 8

LiuZhi1an committed:
Resolve conflicts

LiuZhi1an committed:
Fix bugs

LiuZhi1an committed:
Add new classes for more OOP

LiuZhi1an committed:
Complete More-OOP stretch goal

LiuZhi1an committed:
A-Packages

LiuZhi1an committed:
A-JUnit

LiuZhi1an committed:
A-Jar

LiuZhi1an committed:
Level-9

LiuZhi1an committed:
Add javadoc

LiuZhi1an committed:
A-CodingStandard

LiuZhi1an committed:
A-CodingStandard

LiuZhi1an committed:
Merge tag 'A-CodingStandard'

A-CodingStandard

LiuZhi1an committed:
Merge tag 'Level-9'

Level-9

LiuZhi1an committed:
set up gradle

LiuZhi1an committed:
set up plugins

trash-bin99 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,12 @@
+public class DukeException extends Exception{
+    protected String message;
+
+    /** Constructs a new duke exception with the specified detail message.
+     *
```
Most likely an accident, the JavaDocs description should be 1 line after the initial /** .

trash-bin99 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,82 @@
+     */
+    public static Command parse(String userInput) throws DukeException {
+        int firstBlankIndex = userInput.indexOf(" ");
+        String commandWord = firstBlankIndex == -1 ? userInput : userInput.substring(0, firstBlankIndex);
```
This format is completely valid and compact. I think it would be easier for others to understand your code
if you expanded this statement, to something like:

```

String commandWord;
if (firstBlankIndex == -1) {
    commandWord = userInput;
} else {
    commandWord = userInput.substring(0, firstBlankIndex);
}
```

BananaTechs replied:
I think this format would also have a similar effect.
```
String commandWord = firstBlankIndex == -1
    ? userInput
    : userInput.substring(0, firstBlankIndex);
```

trash-bin99 commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,89 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task{
```
According to the coding standards, we need descriptive header comments for all public classes as well as their methods. I noticed this with other public classses as well.

trash-bin99 reviewed (commented):
Hello!! After looking through your code, I found very little coding standard violations, and I also liked how everything was neatly packaged into different classes (e.g. the Command subclasses). I do think that adding some comments (especially for more complicated parts such as your parse() method) would greatly help both you and others reading your code to understand and spot potential errors. All in all, great job!

LiuZhi1an committed:
implement checkstyle

LiuZhi1an committed:
part1

LiuZhi1an committed:
JavaFX-part2

LiuZhi1an committed:
JavaFX-part3

LiuZhi1an committed:
JavaFX-part-4

LiuZhi1an committed:
create gui file

LiuZhi1an committed:
add GUi to Duke

LiuZhi1an committed:
edited javadoc

LiuZhi1an committed:
temp3

BananaTechs commented on src/main/java/duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,69 @@
+    private String addTask(Task task, TaskList tasks) {
+        String message1 = ("Got it. I've added this task: \n");
+        String message2 = task.toString();
+        tasks.getTaskList().add(task);
```
Should a method like add() could be added to the TaskList class instead of asking it for the ArrayList inside each time? It would better follow the tell-don't-ask principle.

BananaTechs commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,28 @@
+        this.taskList = new ArrayList<>();
+    }
+
+    public ArrayList<Task> getTaskList() {
```
Perhaps this class could have additional functionality to operate on the ArrayList and avoid having to return it?

BananaTechs commented on src/main/java/duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,69 @@
+            storage.saveTaskToFile(task.toString() + "\n");
+            return addTask(this.task, tasks);
+        } else if (this.task instanceof Deadline) {
+            Deadline deadline = (Deadline) this.task;
```
Is this casting needed? this.task remains as type task, and when it is passed to addTask() it is treated of type Task.

BananaTechs commented on src/main/java/duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,69 @@
+     *
+     * @param task the task to be added
+     */
+    private String addTask(Task task, TaskList tasks) {
```
Perhaps this could use this.task instead of taking in a Task argument?

BananaTechs reviewed (commented):
I like how neat and organised your code is! I just suggested some minor optimisations to the TaskList and AddCommand classes.

LiuZhi1an committed:
create jar using gradle

LiuZhi1an committed:
Add assertions to various files

There are several breakpoints that need to be checked
to ensure the state of the flow is correct.

Thus assertions were added to
Storage.load(), Parser.parse(),
AddCommand.execute(), DeleteCommand.execute()

LiuZhi1an committed:
Improve code quality

*add javadoc comments for all class headers
*remove redundant functions
*rename variable names to follow convention
*remove functions no longer used after changing to GUi

LiuZhi1an committed:
use stream in finding tasks

LiuZhi1an committed:
Merge pull request #1 from LiuZhi1an/branch-A-CodeQuality

Merge Branch A-CodeQuality

LiuZhi1an committed:
resolve conflicts

LiuZhi1an committed:
resolve conflicts for stream branch

LiuZhi1an committed:
Merge pull request #2 from LiuZhi1an/branch-A-Streams

Branch A-Streams

LiuZhi1an committed:
Merge pull request #3 from LiuZhi1an/branch-A-Assertions

Branch A-Assertions

LiuZhi1an committed:
add contact functionality

add "add contact" and "delete contact"

LiuZhi1an committed:
fixed bugs about contact

LiuZhi1an committed:
add list and search to contacts

LiuZhi1an committed:
exit window when inputting bye

LiuZhi1an committed:
refine duke exception for GUi

LiuZhi1an committed:
refine duke exception for GUi

LiuZhi1an committed:
give credit for reusing JavaFX tutorial

LiuZhi1an committed:
add user guide

LiuZhi1an committed:
apply checkstyle

LiuZhi1an committed:
make final change
